In [116]:
import pandas as pd
import seaborn as sns
sns.get_dataset_names()

['anagrams',
 'anscombe',
 'attention',
 'brain_networks',
 'car_crashes',
 'diamonds',
 'dots',
 'dowjones',
 'exercise',
 'flights',
 'fmri',
 'geyser',
 'glue',
 'healthexp',
 'iris',
 'mpg',
 'penguins',
 'planets',
 'seaice',
 'taxis',
 'tips',
 'titanic']

In [117]:
# Load dataset

df= sns.load_dataset('titanic')
df.head()

,survived,pclass,sex,age,sibsp,parch,fare,embarked,class,who,adult_male,deck,embark_town,alive,alone
0,0,3,male,22.0,1,0,7.2500,S,Third,man,True,NaN,Southampton,no,False
1,1,1,female,38.0,1,0,71.2833,C,First,woman,False,C,Cherbourg,yes,False
2,1,3,female,26.0,0,0,7.9250,S,Third,woman,False,NaN,Southampton,yes,True
3,1,1,female,35.0,1,0,53.1000,S,First,woman,False,C,Southampton,yes,False
4,0,3,male,35.0,0,0,8.0500,S,Third,man,True,NaN,Southampton,no,True


In [118]:
df.shape

(891, 15)

There is 891 rows and 15 columns.

# Column's description
survived: survival (0 = No, 1 = Yes)
pclass:  Ticket class (1st, 2nd, 3rd)
age: Age in years
sibsp: Number of siblings/spouses
parch: :No. of parents / children aboard the Titanic
fare: Ticket price.
embarked: Port of Embarkation (C = Cherbourg, Q = Queenstown, S = Southampton)
clas: Same as pclass (First, Second, Third)
who: Categorization of passengers as man, woman, or child
adult_male: Boolean indicating if the passenger is an adult male
deck: The deck level where the passenger cabin was located
embark_town: The full name of the town of embarkation.
alive: Descriptive version of survived (yes/no)
alone: Boolean indicating if the passenger was traveling without family.



The target column of this dataset is survived and others are independent feature

Data Analysis

In [119]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 891 entries, 0 to 890
Data columns (total 15 columns):
 #   Column       Non-Null Count  Dtype   
---  ------       --------------  -----   
 0   survived     891 non-null    int64   
 1   pclass       891 non-null    int64   
 2   sex          891 non-null    object  
 3   age          714 non-null    float64 
 4   sibsp        891 non-null    int64   
 5   parch        891 non-null    int64   
 6   fare         891 non-null    float64 
 7   embarked     889 non-null    object  
 8   class        891 non-null    category
 9   who          891 non-null    object  
 10  adult_male   891 non-null    bool    
 11  deck         203 non-null    category
 12  embark_town  889 non-null    object  
 13  alive        891 non-null    object  
 14  alone        891 non-null    bool    
dtypes: bool(2), category(2), float64(2), int64(4), object(5)
memory usage: 80.7+ KB


In [120]:
df.isna().sum()

survived         0
pclass           0
sex              0
age            177
sibsp            0
parch            0
fare             0
embarked         2
class            0
who              0
adult_male       0
deck           688
embark_town      2
alive            0
alone            0
dtype: int64

We found 177 NaN value for age column,
embark_town and embark have 2 Nan,
deck have 688 Nan which is the highest

In [121]:
df1 = df.copy()
df2 = df.copy()
df1.head(10)

,survived,pclass,sex,age,sibsp,parch,fare,embarked,class,who,adult_male,deck,embark_town,alive,alone
0,0,3,male,22.0,1,0,7.2500,S,Third,man,True,NaN,Southampton,no,False
1,1,1,female,38.0,1,0,71.2833,C,First,woman,False,C,Cherbourg,yes,False
2,1,3,female,26.0,0,0,7.9250,S,Third,woman,False,NaN,Southampton,yes,True
3,1,1,female,35.0,1,0,53.1000,S,First,woman,False,C,Southampton,yes,False
4,0,3,male,35.0,0,0,8.0500,S,Third,man,True,NaN,Southampton,no,True
5,0,3,male,NaN,0,0,8.4583,Q,Third,man,True,NaN,Queenstown,no,True
6,0,1,male,54.0,0,0,51.8625,S,First,man,True,E,Southampton,no,True
7,0,3,male,2.0,3,1,21.0750,S,Third,child,False,NaN,Southampton,no,False
8,1,3,female,27.0,0,2,11.1333,S,Third,woman,False,NaN,Southampton,yes,False
9,1,2,female,14.0,1,0,30.0708,C,Second,child,False,NaN,Cherbourg,yes,False


In [122]:
df1.columns
df1 = df1.drop('class', axis= 1)
df1.head()

,survived,pclass,sex,age,sibsp,parch,fare,embarked,who,adult_male,deck,embark_town,alive,alone
0,0,3,male,22.0,1,0,7.2500,S,man,True,NaN,Southampton,no,False
1,1,1,female,38.0,1,0,71.2833,C,woman,False,C,Cherbourg,yes,False
2,1,3,female,26.0,0,0,7.9250,S,woman,False,NaN,Southampton,yes,True
3,1,1,female,35.0,1,0,53.1000,S,woman,False,C,Southampton,yes,False
4,0,3,male,35.0,0,0,8.0500,S,man,True,NaN,Southampton,no,True


In [123]:
df1.columns
df1 = df1.drop('embarked', axis= 1)
df1.head()

,survived,pclass,sex,age,sibsp,parch,fare,who,adult_male,deck,embark_town,alive,alone
0,0,3,male,22.0,1,0,7.2500,man,True,NaN,Southampton,no,False
1,1,1,female,38.0,1,0,71.2833,woman,False,C,Cherbourg,yes,False
2,1,3,female,26.0,0,0,7.9250,woman,False,NaN,Southampton,yes,True
3,1,1,female,35.0,1,0,53.1000,woman,False,C,Southampton,yes,False
4,0,3,male,35.0,0,0,8.0500,man,True,NaN,Southampton,no,True


In [124]:
df1 = df1.drop('alive', axis= 1)
df1.head()

,survived,pclass,sex,age,sibsp,parch,fare,who,adult_male,deck,embark_town,alone
0,0,3,male,22.0,1,0,7.2500,man,True,NaN,Southampton,False
1,1,1,female,38.0,1,0,71.2833,woman,False,C,Cherbourg,False
2,1,3,female,26.0,0,0,7.9250,woman,False,NaN,Southampton,True
3,1,1,female,35.0,1,0,53.1000,woman,False,C,Southampton,False
4,0,3,male,35.0,0,0,8.0500,man,True,NaN,Southampton,True


In [125]:
df1.duplicated().sum()

np.int64(107)

We will remove the duplicates

In [126]:
df1 = df1.drop_duplicates()
df1.duplicated().sum()

np.int64(0)

In [127]:
df1['survived'].value_counts()

survived
0    461
1    323
Name: count, dtype: int64

461 people died, 
323 people survived

In [128]:
df1['embark_town'].value_counts()

embark_town
Southampton    568
Cherbourg      155
Queenstown      59
Name: count, dtype: int64

Most people get in the ship from southampton it is 568, 
Least people was from Queenstown and from cherbourg 155 people get in the ship.

In [129]:
df1.columns
df1['sex'].value_counts()

sex
male      491
female    293
Name: count, dtype: int64

Males passegers were more than the female passengers

In [130]:
df1.columns

Index(['survived', 'pclass', 'sex', 'age', 'sibsp', 'parch', 'fare', 'who',
       'adult_male', 'deck', 'embark_town', 'alone'],
      dtype='object')

In [131]:
df1['pclass'].value_counts()

pclass
3    405
1    214
2    165
Name: count, dtype: int64

Most of the passenger took third class ticket. 
214 people took first class and 165 people took second class ticket

In [132]:
df1['sibsp'].value_counts()

sibsp
0    517
1    201
2     27
4     18
3     14
5      5
8      2
Name: count, dtype: int64

Most of the people were alone. A remarked amount of people was travelling dual.

In [133]:
df1['adult_male'].value_counts()

adult_male
True     451
False    333
Name: count, dtype: int64

451 male was adult that means young. 
333 male was not adult. Here women also count as false. So womens are also count as not adult men.

In [134]:
df1['alone'].value_counts()

alone
True     446
False    338
Name: count, dtype: int64

446 people were travelling alone.

In [135]:
df1.columns

Index(['survived', 'pclass', 'sex', 'age', 'sibsp', 'parch', 'fare', 'who',
       'adult_male', 'deck', 'embark_town', 'alone'],
      dtype='object')

In [136]:
df1.groupby('pclass')['survived'].sum()

pclass
1    135
2     84
3    104
Name: survived, dtype: int64

Total 323 people survived means (survived = 1). Here 135 people from first class. 84 people from second class and 104 people from the third class.

In [137]:
df1['survived'].value_counts()

survived
0    461
1    323
Name: count, dtype: int64

In [138]:
df1.groupby('sex')['survived'].sum()

sex
female    217
male      106
Name: survived, dtype: int64

In [139]:
df1.groupby('embark_town')['survived'].sum()

embark_town
Cherbourg       90
Queenstown      20
Southampton    211
Name: survived, dtype: int64

217 Female survived the accident. On the other hand, 106 Male survived the accident

In [140]:
df1.columns

Index(['survived', 'pclass', 'sex', 'age', 'sibsp', 'parch', 'fare', 'who',
       'adult_male', 'deck', 'embark_town', 'alone'],
      dtype='object')

In [141]:
df1.groupby('pclass')['survived'].value_counts()

pclass  survived
1       1           135
        0            79
2       1            84
        0            81
3       0           301
        1           104
Name: count, dtype: int64

In [142]:
df1.groupby('sex')['survived'].value_counts()

sex     survived
female  1           217
        0            76
male    0           385
        1           106
Name: count, dtype: int64

In [143]:
df1.groupby('embark_town')['survived'].value_counts()

embark_town  survived
Cherbourg    1            90
             0            65
Queenstown   0            39
             1            20
Southampton  0           357
             1           211
Name: count, dtype: int64

In [144]:
df1['died'] = 1 - df1['survived'] #create a new column named died
df1.head()

,survived,pclass,sex,age,sibsp,parch,fare,who,adult_male,deck,embark_town,alone,died
0,0,3,male,22.0,1,0,7.2500,man,True,NaN,Southampton,False,1
1,1,1,female,38.0,1,0,71.2833,woman,False,C,Cherbourg,False,0
2,1,3,female,26.0,0,0,7.9250,woman,False,NaN,Southampton,True,0
3,1,1,female,35.0,1,0,53.1000,woman,False,C,Southampton,False,0
4,0,3,male,35.0,0,0,8.0500,man,True,NaN,Southampton,True,1


Data Cleaning.

We will clean Nan value. Duplicates

In [145]:
df3 = df1.copy()
df4 = df1.copy()
df3

,survived,pclass,sex,age,sibsp,parch,fare,who,adult_male,deck,embark_town,alone,died
0,0,3,male,22.0,1,0,7.2500,man,True,NaN,Southampton,False,1
1,1,1,female,38.0,1,0,71.2833,woman,False,C,Cherbourg,False,0
2,1,3,female,26.0,0,0,7.9250,woman,False,NaN,Southampton,True,0
3,1,1,female,35.0,1,0,53.1000,woman,False,C,Southampton,False,0
4,0,3,male,35.0,0,0,8.0500,man,True,NaN,Southampton,True,1
...,...,...,...,...,...,...,...,...,...,...,...,...,...
885,0,3,female,39.0,0,5,29.1250,woman,False,NaN,Queenstown,False,1
887,1,1,female,19.0,0,0,30.0000,woman,False,B,Southampton,True,0
888,0,3,female,NaN,1,2,23.4500,woman,False,NaN,Southampton,False,1
889,1,1,male,26.0,0,0,30.0000,man,True,C,Cherbourg,True,0


In [146]:
df[df['who']=='child']

,survived,pclass,sex,age,sibsp,parch,fare,embarked,class,who,adult_male,deck,embark_town,alive,alone
7,0,3,male,2.00,3,1,21.0750,S,Third,child,False,NaN,Southampton,no,False
9,1,2,female,14.00,1,0,30.0708,C,Second,child,False,NaN,Cherbourg,yes,False
10,1,3,female,4.00,1,1,16.7000,S,Third,child,False,G,Southampton,yes,False
14,0,3,female,14.00,0,0,7.8542,S,Third,child,False,NaN,Southampton,no,True
16,0,3,male,2.00,4,1,29.1250,Q,Third,child,False,NaN,Queenstown,no,False
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
831,1,2,male,0.83,1,1,18.7500,S,Second,child,False,NaN,Southampton,yes,False
850,0,3,male,4.00,4,2,31.2750,S,Third,child,False,NaN,Southampton,no,False
852,0,3,female,9.00,1,1,15.2458,C,Third,child,False,NaN,Cherbourg,no,False
869,1,3,male,4.00,1,1,11.1333,S,Third,child,False,NaN,Southampton,yes,False


In [147]:
# Nan value remove

df1.isna().sum()

survived         0
pclass           0
sex              0
age            106
sibsp            0
parch            0
fare             0
who              0
adult_male       0
deck           582
embark_town      2
alone            0
died             0
dtype: int64

In [148]:
df1.shape

(784, 13)

In [149]:
df3.dropna(inplace= True)
df3.shape

(181, 13)

In [150]:
df3.duplicated().sum()

np.int64(0)

In [151]:
age_mean = df1['age'].mean()
age_mean

df1['age'] = df1['age'].fillna(age_mean)
df1['age'].isna().sum()
df1.isna().sum()

survived         0
pclass           0
sex              0
age              0
sibsp            0
parch            0
fare             0
who              0
adult_male       0
deck           582
embark_town      2
alone            0
died             0
dtype: int64

The Nan in age was removed by the comlun's mean.

In [152]:
# embark_town nan removal 

emb_town_mode = df1['embark_town'].mode()

In [153]:
df1['embark_town'] = df1['embark_town'].fillna(emb_town_mode)[0]
df1.isna().sum()

survived         0
pclass           0
sex              0
age              0
sibsp            0
parch            0
fare             0
who              0
adult_male       0
deck           582
embark_town      0
alone            0
died             0
dtype: int64

Deck has a lot of Nan value.
So we will remove the column.

In [154]:
df1.columns
df1 = df1.drop('deck', axis= 1)
df1.head()

,survived,pclass,sex,age,sibsp,parch,fare,who,adult_male,embark_town,alone,died
0,0,3,male,22.0,1,0,7.2500,man,True,Southampton,False,1
1,1,1,female,38.0,1,0,71.2833,woman,False,Southampton,False,0
2,1,3,female,26.0,0,0,7.9250,woman,False,Southampton,True,0
3,1,1,female,35.0,1,0,53.1000,woman,False,Southampton,False,0
4,0,3,male,35.0,0,0,8.0500,man,True,Southampton,True,1
